lightning/src/sync/mod.rs

```diff
@@ -1,3 +1,15 @@
+#[allow(dead_code)] // Depending on the compilation flags some variants are never used
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) enum LockHeldState {
+	HeldByThread,
+	NotHeldByThread,
+	Unsupported,
```

Comment:
I think a comment explaining when a lock held state cannot be determined would be helpful